src/NHibernate/Transform/AliasToBeanResultTransformer.cs

```diff
@@ -111,54 +80,26 @@ public override IList TransformList(IList collection)
 
 		#region Setter resolution
 
-		/// <summary>
-		/// Set the value of a property or field matching an alias.
-		/// </summary>
-		/// <param name="alias">The alias for which resolving the property or field.</param>
-		/// <param name="value">The value to which the property or field should be set.</param>
-		/// <param name="resultObj">The object on which to set the property or field. It must be of the type for which
-		/// this instance has been built.</param>
-		/// <exception cref="PropertyNotFoundException">Thrown if no matching property or field can be found.</exception>
-		/// <exception cref="AmbiguousMatchException">Thrown if many matching properties or fields are found, having the
-		/// same visibility and inheritance depth.</exception>
-		private void SetProperty(string alias, object value, object resultObj)
-		{
-			if (alias == null)
-				// Grouping properties in criteria are selected without alias, just ignore them.
-				return;
-
-			if (TrySet(alias, value, resultObj, _propertiesByNameCaseSensitive))
-				return;
-			if (TrySet(alias, value, resultObj, _fieldsByNameCaseSensitive))
-				return;
-			if (TrySet(alias, value, resultObj, _propertiesByNameCaseInsensitive))
-				return;
-			if (TrySet(alias, value, resultObj, _fieldsByNameCaseInsensitive))
-				return;
-
-			throw new PropertyNotFoundException(resultObj.GetType(), alias, "setter");
-		}
-
-		private bool TrySet(string alias, object value, object resultObj, Dictionary<string, NamedMember<FieldInfo>> fieldsMap)
+		protected MemberInfo GetMemberInfo(string alias)
 		{
-			if (fieldsMap.TryGetValue(alias, out var field))
-			{
-				CheckMember(field, alias);
-				field.Member.SetValue(resultObj, value);
-				return true;
-			}
-			return false;
+			return TryGetMemberInfo(alias, _propertiesByNameCaseSensitive)
+				?? TryGetMemberInfo(alias, _fieldsByNameCaseSensitive)
+				?? TryGetMemberInfo(alias, _propertiesByNameCaseInsensitive)
+				?? (MemberInfo) TryGetMemberInfo(alias, _fieldsByNameCaseInsensitive)
+				?? throw new PropertyNotFoundException(_resultClass, alias, "setter");
 		}
 
-		private bool TrySet(string alias, object value, object resultObj, Dictionary<string, NamedMember<PropertyInfo>> propertiesMap)
+		private TMemberInfo TryGetMemberInfo<TMemberInfo>(string alias, params Dictionary<string, NamedMember<TMemberInfo>>[] propertiesMaps) where TMemberInfo: MemberInfo
```

Comment:
The `params` change seems unused. I guess it is some kind of improvement attempt not cleaned up. Can it be cleaned up?